lib/swarm/distribution/static_quorum_ring.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Swarm.Distribution.StaticQuorumRing do
+  @moduledoc """
+  A quorum is the minimum number of nodes that a distributed cluster has to obtain in order to be
+  allowed to perform an operation. This can be used to enforce consistent operation in a distributed system.
+
+  ## Quorum size
+
+  You must configure the quorum size using the `:static_quorum_size` setting:
+
+      config :swarm,
+        static_quorum_size: 5
+
+  It defines the minimum number of nodes that must be connected in the cluster to allow process
+  registration and distribution.
+
+  If there are fewer nodes currently available than the quorum size, any calls to
```

Comment:
It may be worth discussing here the use of the `kernel` config options `:sync_nodes_mandatory`, `:sync_nodes_optional`, and `:sync_nodes_timeout`. These ensure the required and optional members of the cluster are connected when the runtime boots and before any applications start, it's particularly useful for use cases this strategy is designed around (i.e. the cluster members are known in advance). The `mandatory` and `optional` settings take a list of nodes, and the `timeout` setting takes an integer or `:infinity`. You can configure it like any other app, e.g.:

```elixir
config :kernel,
  sync_nodes_mandatory: [:"node1@192.168.1.1", :"node2@192.168.1.2"],
  sync_nodes_timeout: 60_000
```